### PR TITLE
Update CPM65; add some missing ctype and string functions

### PR DESCRIPTION
**Pull request by davidgiven**
This:

- fixes a bug in the CPM65 headers where one function would corrupt the parameters when called
- updated with some new CPM65 functionality
- added a `cpm_parse_filename` helper function (in pure C)
- adds `isalnum` and `isspace` to ctype
- adds `strrev`, `strspn`, `strpbrk` and `strtok` to strings

### CODE REVIEW

File: mos-platform/common/c/string.c [hunk 1]
```diff
@@ -63,3 +63,70 @@ __attribute__((weak)) char *strrchr(const char *s, int c) {
...
+{
+    size_t len = strlen((const char*)str);
+    size_t i, j;
+    for (i = 0, j = len - 1; i < j; i++, j--)
+    {
+        char a = str[i];
+        str[i] = str[j];
+        str[j] = a;
+    }
+	return str;
```
Comment (mysterymath) [inline mos-platform/common/c/string.c:77]: nit: decrease indent

File: mos-platform/common/c/string.c [hunk 2]
```diff
@@ -63,3 +63,70 @@ __attribute__((weak)) char *strrchr(const char *s, int c) {
       last = s;
   return (char *)last;
 }
+
+__attribute__((weak)) char* strrev(char* str)
+{
+    size_t len = strlen((const char*)str);
+    size_t i, j;
+    for (i = 0, j = len - 1; i < j; i++, j--)
```
Comment (mysterymath) [inline mos-platform/common/c/string.c:71]: nit: prefer declaring variables inside loops if only used inside, here and throughout. The llvm-mos stdlib generally targets c99, and even C++ is allowed if ABI-compatible.

File: mos-platform/common/c/string.c [hunk 3]
```diff
@@ -63,3 +63,70 @@ __attribute__((weak)) char *strrchr(const char *s, int c) {
...
+
+    if (string == NULL)
+    {
+        string = savestring;
+        if (string == NULL)
+            return (char*)NULL;
+    }
+
+    char* s1 = string + strspn(string, separators);
+    if (*s1 == '\0')
```
Comment (mysterymath) [inline mos-platform/common/c/string.c:120]: Use !*s1, for consistency with general use of *s1 as a conditional

**Review by mysterymath — APPROVED** [review]
Hey, thanks for putting this together! Always nice to see more std library functions :)

File: mos-platform/common/c/string.c (hunk 2 above)
Comment (davidgiven) [reply mos-platform/common/c/string.c:71]: I actually had to look up how to declare two variables at once in a for loop. I think I died inside a little. Done.

File: mos-platform/common/c/string.c (hunk 1 above)
Comment (davidgiven) [reply mos-platform/common/c/string.c:77]: Forgot to run through clang-format. Done.

File: mos-platform/common/c/string.c (hunk 3 above)
Comment (davidgiven) [reply mos-platform/common/c/string.c:120]: Done.

**Pull request merged**